hashing/hashing.go

```diff
@@ -0,0 +1,57 @@
+package hashing
+
+import (
+	"fmt"
+	"hash/fnv"
+	"k8s.io/apimachinery/pkg/util/rand"
+	hashutil "k8s.io/kubernetes/pkg/util/hash"
+	"reflect"
+)
+
+func ComputeHashForKubernetesObject(object interface{}) (string, error) {
+	value := reflect.ValueOf(object)
+	if value.Kind() != reflect.Ptr {
+		return "", fmt.Errorf("Must pass pointer to ComputeHashForKubernetesObject!")
+		// This is because the hash function prints the type which will be different
+		// if the object is a pointer (even if they are otherwise identical)
+		// normally we would only accept a pointer but this has to accept an interface
+		// so we can use it with any k8s object.
+	}
+	objectMeta := value.Elem().FieldByName("ObjectMeta")
+	if objectMeta.Kind() == reflect.Struct {
+		labels := objectMeta.FieldByName("Labels")
+		if labels.Kind() == reflect.Map {
+			// equivalent of delete(labels, "yelp.com/operator_config_hash")
+			// we remove this so that the already hashed versions will match
+			// the newly calculated versions
+			labels.SetMapIndex(reflect.ValueOf("yelp.com/operator_config_hash"), reflect.Value{})
+		}
+	}
+	hasher := fnv.New32a()
+	hashutil.DeepHashObject(hasher, object)
+	return rand.SafeEncodeString(fmt.Sprint(hasher.Sum32())), nil
+}
+
+func GetHashFromKubernetesObject(object interface{}) (string, error) {
```

Comment:
I'm thinking maybe this function is overkill? Maybe the caller can just get the hash from the label themselves?

Reply:
Interesting thought! I think a label is good because it gives us the ability to select based on it. I'm basically copying how the ReplicaSet controller works with Pods (and how PaaSTA has worked before kubernetes existed!)

Right now this isn't a big deal (at least for cassandra) because we are just updating the StatefulSet when it changes. However, in a more complex future we might want to handle "bounce logic". Imagine you want to change the `volumeClaimTemplate` but as we found you can't update it after creating a StatefulSet. In theory the Operator could:
* drain all nodes on existing cluster
* trigger a backup
* create new StatefulSet with restore from backup option
* delete old StatefulSet

For anything like this we'll want to be able to select based on the label I think.